rust/src/fvm/types.rs

```diff
@@ -0,0 +1,135 @@
+use std::io::{Error, SeekFrom};
+use std::ptr;
+use std::slice::from_raw_parts;
+
+use anyhow::Result;
+use drop_struct_macro_derive::DropStructMacro;
+use ffi_toolkit::{code_and_message_impl, free_c_str, CodeAndMessage, FCPResponseStatus};
+
+use fvm::message::Message;
+use fvm_shared::address::Address;
+use fvm_shared::clock::ChainEpoch;
+use fvm_shared::econ::TokenAmount;
+use fvm_shared::encoding::RawBytes;
+use fvm_shared::version::NetworkVersion;
+use fvm_shared::MethodNum;
+
+use num_traits::FromPrimitive;
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+pub enum fil_FvmRegisteredVersion {
+    V1,
+}
+
+// Not sure which form is needed
+fn get_address_from_id(id: u64) -> Address {
+    Address::new_id(id)
+}
+
+// Not sure which form is needed
+fn get_address_from_bytes(
+    address_ptr: *const u8,
+    address_len: libc::size_t,
+) -> Result<Address, fvm_shared::address::Error> {
+    let address_bytes: Vec<u8> =
+        unsafe { std::slice::from_raw_parts(address_ptr, address_len).to_vec() };
+
+    Address::from_bytes(&address_bytes)
+}
+
+fn get_params_from_bytes(params_ptr: *const u8, params_len: libc::size_t) -> RawBytes {
+    let params_bytes: Vec<u8> =
+        unsafe { std::slice::from_raw_parts(params_ptr, params_len).to_vec() };
+
+    RawBytes::new(params_bytes)
+}
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+pub struct fil_Message {
+    pub version: u64,
+    pub from_bytes: *const u8,
+    pub from_len: libc::size_t,
+    pub to_bytes: *const u8,
+    pub to_len: libc::size_t,
+    pub sequence: u64,
+    pub value: u64,
+    pub method_num: u64,
+    pub params_bytes: *const u8,
+    pub params_len: libc::size_t,
+    pub gas_limit: u64,
+    pub gas_fee_cap: u64,
+    pub gas_premium: u64,
+}
```

Comment:
We can use a DAG-CBOR-encoded message (which is how it's transmitted in the network anyway), and deserialize it in the Rust-side FFI handler, instead of modelling it again.